src/rendering.rs

```diff
@@ -0,0 +1,182 @@
+use std::ffi::CString;
+use std::num::NonZeroU32;
+use std::rc::Rc;
+
+use euclid::default::Size2D;
+use gleam::gl;
+use glutin::config::GetGlConfig;
+use glutin::context::PossiblyCurrentContext;
+use glutin::prelude::{GlContext, PossiblyCurrentGlContext};
+use glutin::surface::{
+    GlSurface, ResizeableSurface, Surface, SurfaceTypeTrait, SwapInterval, WindowSurface,
+};
+use glutin::{
+    config::{Config, ConfigTemplateBuilder, GlConfig},
+    context::{ContextApi, ContextAttributesBuilder, Version},
+    display::GetGlDisplay,
+    prelude::{GlDisplay, NotCurrentGlContext},
+};
+use glutin_winit::{DisplayBuilder, GlWindow};
+use raw_window_handle::HasWindowHandle;
+use winit::event_loop::ActiveEventLoop;
+use winit::window::Window;
+
+/// A Verso rendering context, which holds all of the information needed
+/// to render Servo's layout, and bridges WebRender and glutin.
+pub struct RenderingContext {
+    context: PossiblyCurrentContext,
+    pub(crate) gl: Rc<dyn gl::Gl>,
+}
+
+impl RenderingContext {
+    /// Create a rendering context instance.
+    pub fn create(
+        evl: &ActiveEventLoop,
+        window: &Window,
+    ) -> Result<(Self, Surface<WindowSurface>), Box<dyn std::error::Error>> {
+        let template = ConfigTemplateBuilder::new()
+            .with_alpha_size(8)
+            .with_transparency(cfg!(macos));
+        let (_, gl_config) = DisplayBuilder::new().build(evl, template, gl_config_picker)?;
+
+        log::debug!("Picked a config with {} samples", gl_config.num_samples());
+
+        // XXX This will panic on Andoird, but we care about Deskyop for now.
+        let raw_window_handle = window.window_handle().ok().map(|handle| handle.as_raw());
+        // XXX The display could be obtained from any object created by it, so we can
+        // query it from the config.
+        let gl_display = gl_config.display();
+        // The context creation part.
+        let context_attributes = ContextAttributesBuilder::new().build(raw_window_handle);
+        // Since glutin by default tries to create OpenGL core context, which may not be
+        // present we should try gles.
```

Comment:
```suggestion
        // present we should try GLES.
```
minor, but since GLES is then mentioned in full capital then might as well use consistent wording.